code/game/objects/items/weapons/table_rack_parts.dm

```diff
@@ -48,13 +49,16 @@
 			to_chat(user, "<span class='notice'>You add glass panes to \the [name].</span>")
 			glass.use(1)
 			qdel(src)
-
 /obj/item/weapon/table_parts/attack_self(mob/user)
-	new /obj/structure/table(user.loc)
+	for(var/atom/movable/AM in get_turf(user))
```

Comment:
you could just use `for(var/obj/structure/table in get_turf(user))` or `if(locate(/obj/structure/table) in get_turf(user))`

Reply:
Probably could. Its 5:30 my brain went to mush writing this stuff.